pipelines/flux/scripts/longhorn-setup.sh

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+
+set -x
+
+source pipelines/utilities/kubeconfig.sh
+source pipelines/utilities/selinux_workaround.sh
+source pipelines/utilities/install_csi_snapshotter.sh
+source pipelines/utilities/create_aws_secret.sh
+source pipelines/utilities/install_backupstores.sh
+source pipelines/utilities/create_longhorn_namespace.sh
+source pipelines/utilities/flux.sh
+source pipelines/utilities/run_longhorn_test.sh
+
+# create and clean tmpdir
+TMPDIR="/tmp/longhorn"
+mkdir -p ${TMPDIR}
+rm -rf "${TMPDIR}/"
+
+export LONGHORN_NAMESPACE="longhorn-system"
+export LONGHORN_INSTALL_METHOD="flux"
+export HELM_CHART_DEFAULT_URL="https://charts.longhorn.io"
+
+
+main(){
+  set_kubeconfig
+
+  if [[ ${DISTRO} == "rhel" ]] || [[ ${DISTRO} == "rockylinux" ]] || [[ ${DISTRO} == "oracle" ]]; then
+    apply_selinux_workaround
+  fi
+
+  # set debugging mode off to avoid leaking aws secrets to the logs.
+  # DON'T REMOVE!
+  set +x
+  create_aws_secret
+  set -x
+
+  create_longhorn_namespace
+  install_backupstores
+  install_csi_snapshotter
+
+  init_flux
+
+  if [[ "${LONGHORN_UPGRADE_TEST}" == true ]]; then
+    create_flux_helm_repo "${HELM_CHART_DEFAULT_URL}"
+    create_flux_helm_release "${LONGHORN_STABLE_VERSION}"
+    LONGHORN_UPGRADE_TYPE="from_stable"
+    LONGHORN_UPGRADE_TEST_POD_NAME="longhorn-test-upgrade-from-stable"
+    if [[ -n "${LONGHORN_TRANSIENT_VERSION}" ]]; then
+      FLUX_HELM_CHART_URL="${HELM_CHART_DEFAULT_URL}"
+      FLUX_HELM_CHART_VERSION="${LONGHORN_TRANSIENT_VERSION}"
+      UPGRADE_LH_ENGINE_IMAGE="longhornio/longhorn-engine:${LONGHORN_TRANSIENT_VERSION}"
+      run_longhorn_upgrade_test
+      LONGHORN_UPGRADE_TYPE="from_transient"
+      LONGHORN_UPGRADE_TEST_POD_NAME="longhorn-test-upgrade-from-transient"
+    fi
+    FLUX_HELM_CHART_URL="${HELM_CHART_URL}"
+    FLUX_HELM_CHART_VERSION="${LONGHORN_INSTALL_VERSION}"
+    UPGRADE_LH_ENGINE_IMAGE="longhornio/longhorn-engine:${LONGHORN_INSTALL_VERSION}"
+    run_longhorn_upgrade_test
+    run_longhorn_test
```

Comment:
Does this mean we don't use flux to do the upgrade testing but rather relying on `run_longhorn_upgrade_test`?

Reply:
`run_longhorn_upgrade_test` is just a function, it applies the `test.yaml` with desired configuration and starts the test pod, and the test pod is who performing the upgrade test in `pytest`, and `pytest`'s upgrade test case will determine which method should be used to perform the upgrade according to the configuration. In this case, it's the configuration for `flux`, so `pytest`'s upgrade test case will call the `flux` related shell script to perform some `flux` operations.

Other upgrade methods do the same thing. The only difference is the configuration.